source/development/coredump.rst

```diff
@@ -0,0 +1,196 @@
+.. Copyright (C) 2024, Wazuh, Inc.
+
+.. meta::
+   :description: This section contains instructions to configure and collect core dumps for analysis.
+
+Configuring core dump generation
+================================
+
+A *core dump* or *crash dump* is a snapshot of a process's memory taken when a serious or unhandled error occurs. The operating system on a monitored endpoint can automatically generate core dumps. These dumps are valuable for diagnosing hanging processes. Alongside environment information such as the operating system version, they can offer insights into the cause of a crash.
+
+Linux endpoints
+---------------
+
+In Linux version 2.41 and later, a template defines the location and name of the generated `core dump files <https://man7.org/linux/man-pages/man5/core.5.html>`__. Earlier versions generated the core dump files next to the location of the file that caused the error.
+
+Using `systemd`
+^^^^^^^^^^^^^^^
+
+To set up core dump generation using systemd, you can utilize systemd's built-in features for managing core dumps. Here's a step-by-step guide:
+
+#. **Check Core Dump Configuration**:
+    First, verify the current core dump configuration:
+
+    .. code-block:: console
+
+        systemctl status systemd-coredump*
+
+    .. code-block:: none                                                                                                                                                                                            
+        :class: output                            
+
+        ● systemd-coredump.socket - Process Core Dump Socket
+             Loaded: loaded (/lib/systemd/system/systemd-coredump.socket; static)
+             Active: active (listening) ...
+
+#. **Identify Target Directory for Core Dumps**:
+    Choose a directory where core dump files will be generated. By default, systemd stores core dump files in ``/var/lib/systemd/coredump/``.
+
+    .. code-block:: console
+
+        cat /proc/sys/kernel/core_pattern
+
+    .. code-block:: none                                                                                                                                                                                            
+        :class: output                            
+
+        │|/lib/systemd/systemd-coredump %P %u %g %s %t                                                                                                                                                                                                                            
+
+#. **Enable Core Dump Collection**:
+    Enable core dump collection by setting the ``Storage=`` option in the systemd ``coredump.conf`` file. You can set it to ``external`` to store core dumps externally, or ``none`` to disable core dump collection altogether.
+
+    .. code-block:: console
+
+        systemctl edit systemd-coredump
+
+    Add the following lines to the editor that opens:
```

Comment:
```suggestion
#. Add the following lines in the editor that opens to enable core dump collection and store core dumps externally. To disable core dump generation, set ``Storage=none``.
```

Reply:
this task description is parto of point `Enable Core Dump Collection`

Reply:
Which I suggested to remove.